File: Actividad_1_Clases_Funciones_Herencia_y_Condicionales_Java/src/animales/Ave.java
package animales;

import java.time.Year;

// Clase Ave que hereda de Animal
public class Ave extends Animal {
    private int añoNacimiento;
    private String propietario; // Se agrega el atributo propietario

    // Constructor que incluye el propietario
    public Ave(String nombre, double peso, int añoNacimiento, String propietario) {
        super(nombre, peso);
        this.añoNacimiento = añoNacimiento;
        this.propietario = propietario;
    }

    // Método para calcular la edad y determinar si es mayor o menor de edad
    public void calcularEdad() {
        int añoActual = Year.now().getValue();
        int edad = añoActual - añoNacimiento;
        
        System.out.println("Nombre: " + nombre + ", Edad: " + edad + " años, Propietario: " + propietario);
        if (edad >= 5) {
            System.out.println("MAYOR DE EDAD");
        } else {
            System.out.println("MENOR DE EDAD");
        }
    }
}

File: Actividad_1_Clases_Funciones_Herencia_y_Condicionales_Pyton/Class/Ave.py
from datetime import datetime
from Class.Animal import Animal  # Importamos la clase Animal

class Ave(Animal):
    def __init__(self, nombre, peso, ano_nacimiento, propietario):
        super().__init__(nombre, peso) 
        self.ano_nacimiento = ano_nacimiento
        self.propietario = propietario

    def calcular_edad(self):
        ano_actual = datetime.now().year
        return ano_actual - self.ano_nacimiento

    def mostrar_edad(self):
        edad = self.calcular_edad()
        if edad >= 5:
            print(f"El ave tiene {edad} años y es MAYOR DE EDAD")
        else:
            print(f"El ave tiene {edad} años y es MENOR DE EDAD")